java/src/com/ibm/streamsx/rest/primitives/InstancesArray.java

```diff
@@ -12,13 +12,15 @@
 import com.ibm.streamsx.rest.primitives.Instance;
 import com.ibm.streamsx.rest.primitives.InstanceGson;
 
+/**
+ * class to hold information about instances from GET instances URL
+ * this class is public, but should only be used by classes in the rest package
```

Comment:
as it is, the primitives are a sub-directory currently.  If I move them up a level, then yes, they can be made package private.  I've only exposed the one InstanceArray as public right now, since the Connection objects need this as the entry point.   Let me know if this is what you want, and I'll make the changes for it.